Adding flatpak

gdeusdara opened the pull request.

bcribas commented on debian/control:
```diff
@@ -6,7 +6,7 @@ Build-Depends: debhelper
 Package: maratona-linguagens
 Architecture: all
-Depends: maratona-snap-common, build-essential, default-jre, default-jdk, libstdc++-7-dev, libstdc++6-7-dbg, gdb, python3, pyflakes, pyflakes3, python
+Depends: maratona-flatpak-common, build-essential, default-jre, default-jdk, libstdc++-7-dev, libstdc++6-7-dbg, gdb, python3, pyflakes, pyflakes3, python, unzip
```
Qual o motivo do pacote maratona-linguagens depender do maratona-flatpak-common?

bcribas commented on debian/maratona-linguagens.postinst:
```diff
@@ -1,28 +1,16 @@
-while [ $flag -eq 1 ]; do
-	packages=""
-	try_install_snap kotlin --classic || packages=$packages"kotlin"
+wget -O /opt/kotlin.zip $URL
```
Além do `wget` é adequado verificar o MD5SUM (ou SHA512SUM) do arquivo que foi baixado, para garantir que foi feito o download do que esperamos.

E também o arquivo deveria ser removido depois de feito o download.

bcribas commented on debian/maratona-vscode-extensions.postinst:
```diff
@@ -55,3 +55,6 @@ while $flag ; do
 done
+
+rm -rf /usr/share/maratona-vscode-extensions/*
+cp -r /opt/vscode-extensions/* /usr/share/maratona-vscode-extensions/ && rm -rf /opt/vscode-extensions/
```
mv /opt/vscode-extensions /usr/share/maratona-vscode-extensions

bcribas commented on debian/maratona-vscode-extensions.postinst:
```diff
@@ -55,3 +55,6 @@ while $flag ; do
 	fi
 done
+
+rm -rf /usr/share/maratona-vscode-extensions/*
```
rm -rf /usr/share/maratona-vscode-extensions

bcribas commented:
Outro ponto importante que está sendo deixado de lado (mas que pode ser resolvido nos próximos commits) é o funcionamento dos pacotes quanto ao remove e ao upgrade.

Por exemplo, no pacote maratona-vscode-extensions, em um eventual upgrade na versão do pacote ele baixará tudo novamente, mesmo as extensões que não foram atualizadas, será que esse é o comportamento esperado?

gdeusdara committed:
installing kotlin from url

Antes, o kotlin era instalado via snap, mas não usaremos mais o snapd
para instalação de pacotes. Portanto, foi preciso migrar essa instalação
do kotlin para a instalação padrão

gdeusdara committed:
switching from snapd to flatpak
